kernel/procfs.c

```diff
@@ -574,6 +574,33 @@ static int _pid_stat_vcallback(
     return ret;
 }
 
+static int _sys_vcallback(
+    myst_file_t* self,
+    myst_buf_t* vbuf,
+    const char* entrypath)
+{
+    (void)self;
+    int ret = 0;
+
+    (void)entrypath;
+
+    if (!vbuf)
+        ERAISE(-EINVAL);
+
+    myst_buf_clear(vbuf);
+    char tmp[128];
+
+    ECHECK(myst_snprintf(tmp, sizeof(tmp), "32768\n"));
+    ECHECK(myst_buf_append(vbuf, tmp, strlen(tmp)));
+
+done:
+
+    if (ret != 0)
+        myst_buf_release(vbuf);
```

Comment:
This buffer was constructed by the caller and so should be destructed by the function. I would remove this.